From: Alberto Garcia <berto@igalia.com>
Bug: https://bugs.webkit.org/show_bug.cgi?id=172799
Description: Use CMAKE_CXX_LIBRARY_ARCHITECTURE to detect x86
Adapted for 2.24.4
----
--- a/CMakeLists.txt.orig	2019-08-26 07:55:41.000000000 -0700
+++ b/CMakeLists.txt	2019-09-25 20:07:39.634458507 -0700
@@ -92,6 +92,10 @@ elseif (LOWERCASE_CMAKE_SYSTEM_PROCESSOR
     set(WTF_CPU_MIPS64 1)
 elseif (LOWERCASE_CMAKE_SYSTEM_PROCESSOR MATCHES "^mips")
     set(WTF_CPU_MIPS 1)
+elseif (CMAKE_CXX_LIBRARY_ARCHITECTURE MATCHES "(x64|x86_64|amd64)")
+    set(WTF_CPU_X86_64 1)
+elseif (CMAKE_CXX_LIBRARY_ARCHITECTURE MATCHES "(i[3-6]86|x86)")
+    set(WTF_CPU_X86 1)
 elseif (LOWERCASE_CMAKE_SYSTEM_PROCESSOR MATCHES "(x64|x86_64|amd64)")
     set(WTF_CPU_X86_64 1)
 elseif (LOWERCASE_CMAKE_SYSTEM_PROCESSOR MATCHES "(i[3-6]86|x86)")
